### ISSUE TITLE
Clean Up Python Warnings

### ISSUE THREAD

**Issue by jtpereyda** [OpenRCE/sulley#84]
This proposal is to basically use Pycharm's refactor features to eliminate all/most style warnings. There would be a lot of changes, but all relatively minor.

I'm ready to do this myself, as it would help ongoing development.

If sulley_refactor gets merged back in later, this shouldn't create too much churn as the diffs should collate more or less.

Any thoughts?

**Comment by Fitblip** [comment]
Hey @jtpereyda,

I actually already did this :), that's the sulley_refactor branch (as well as a few other features like flask/tornado vs BaseHTTPServer). The plan was to go in and write legit tests for everything, then re-merge to master for sulley v2.0.0. 

Thoughts?

**Comment by jtpereyda** [comment]
Hmm... Do you think it's good enough to merge in for now, before the tests? I'd love to have access to those changes going forward.

Is `sulley_refactor` good enough to run off of? If so, one option is for me to start working off of the branch.

**Comment by jtpereyda** [comment]
Also, what's the to do list for sulley_refactor? Just unit tests? What can I do to help?

**Comment by Fitblip** [comment]
Yeah I definitely think sulley_refactor is good enough to work off of TBH. Things may be varying levels of broken (since we have no usable tests :(), but for the most part it should just be cleaned up and have a few extra things. 

The todo list is to just test the shit out of everything, so if you're down to work on that a bit it'd be much appreciated :). There may be a few PRs that didn't get merged in (your signal fix is definitely not there).

**Comment by jtpereyda** [comment]
I'll pull it down and explore. Just unit tests, right?

If I can do PRs against sulley_refactor, we can treat it like a working 'develop' branch (http://nvie.com/posts/a-successful-git-branching-model/). That way development won't be blocked by missing tests required for a release. :)

**Comment by Fitblip** [comment]
Yep, unit tests are a really good place to start!

As for PRs - sure! That sounds great.
